src/v/iceberg/tests/schema_protobuf_test.cc

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/datatypes.h"
+#include "iceberg/schema_converter.h"
+#include "iceberg/schema_protobuf.h"
+#include "iceberg/tests/test_schemas.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/core.h>
+#include <fmt/ostream.h>
+#include <gmock/gmock.h>
+#include <google/protobuf/compiler/parser.h>
+#include <google/protobuf/descriptor.h>
+#include <google/protobuf/descriptor.pb.h>
+#include <google/protobuf/io/tokenizer.h>
+#include <gtest/gtest.h>
+using namespace iceberg;
+
+namespace {
+
+void parse_proto_definition_string(
+  const ss::sstring& proto_str, google::protobuf::DescriptorPool& pool) {
+    google::protobuf::compiler::Parser parser;
+
+    google::protobuf::io::ArrayInputStream input_stream(
+      proto_str.c_str(), proto_str.size());
+
+    google::protobuf::io::Tokenizer tokenizer(&input_stream, nullptr);
+    google::protobuf::FileDescriptorProto fdp;
+    fdp.set_name("test_schema_file");
+
+    auto parse_result = parser.Parse(&tokenizer, &fdp);
+    ASSERT_TRUE(parse_result);
+    pool.BuildFile(fdp);
+}
+
+}; // namespace
+
+MATCHER_P3(IsField, id, name, type, "") {
+    fmt::print(
+      "field(id: {}, name: {}, type: {}) expected to be equal to field(id: {}, "
+      "name: {}, type: {})\n",
+      arg->id,
+      arg->name,
+      arg->type,
+      id,
+      name,
+      type);
+    *result_listener << "";
+
+    return arg->id == id && arg->name == name && arg->type == type;
```

Comment:
Just curious what this is?

Reply:
this is gtest matcher i.e. it is a way to express expectations about a given variable like this: 
```
EXPECT_THAT(top_level.fields[0], IsField(0, "name", string_type{}));
```